docarray/typing/tensor/video/video_tensor_mixin.py

```diff
@@ -0,0 +1,118 @@
+from typing import TYPE_CHECKING, BinaryIO, Optional, Type, TypeVar, Union
+
+import numpy as np
+
+from docarray.typing.tensor.audio.audio_tensor import AudioTensor
+
+if TYPE_CHECKING:
+    from docarray.typing import VideoNdArray, VideoTorchTensor
+
+
+T = TypeVar('T', bound='VideoTensorMixin')
+
+
+class VideoTensorMixin:
+    @staticmethod
+    def validate_shape(
+        cls: Union[Type['VideoTorchTensor'], Type['VideoNdArray']], value: 'T'
+    ) -> 'T':
+        comp_backend = cls.get_comp_backend()
+
+        if (
+            comp_backend.n_dim(value) not in [3, 4]  # type: ignore
```

Comment:
why do we need to ignore the type ?

Reply:
I think this is this issue that has to be fixed using the ugly overloads.

Reply:
@JohannesMessner need ur help there we did not manage to debug this. I don't understand why it is not working and why we have different TypeVar in AbstractCompBackend